test/integration/driver.go

```diff
@@ -29,30 +30,41 @@ import (
 	"time"
 
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
 	"k8s.io/client-go/pkg/api/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 
+	"github.com/ghodss/yaml"
 	flag "github.com/spf13/pflag"
+
+	"istio.io/manager/model"
+	"istio.io/manager/platform/kube"
 )
 
 const (
-	yaml = "echo.yaml"
+	managerYaml      = "manager.yaml"
+	simpleAppYaml    = "simple-app.yaml"
+	versionedAppYaml = "versioned-app.yaml"
 
 	// budget is the maximum number of retries with 1s delays
 	budget = 30
 )
 
 var (
-	kubeconfig string
-	hub        string
-	tag        string
-	namespace  string
-	dump       bool
-	client     *kubernetes.Clientset
+	kubeconfig  string
+	hub         string
+	tag         string
+	namespace   string
+	verbose     bool
+	client      *kubernetes.Clientset
+	istioClient *kube.Client
 )
 
-func cleanup() {
+func teardown() {
+	if err := istioClient.DeregisterResources(); err != nil {
```

Comment:
Let's not de-register resources. TPRs definitions are global, so this will break any concurrent test.